### ISSUE TITLE
`pkexec` doesn't work on macOS

### ISSUE THREAD

**Issue by npalladium** [aklos/autotool#18]
`pkexec` (`policykit`/`polkit` in general ) doesn't seem to be installed on macOS by default and there is no hassle-free way to do it afaik.

macOS version is:
```
Darwin 20.6.0 Darwin Kernel Version 20.6.0; root:xnu-7195.141.46~1/RELEASE_X86_64
```

The elevated privilege example in the User manual keeps running and doesn't do anything.

Would it be possible to either:
- Get this to work on macOS?
- Throw an early error?

**Comment by aklos** [comment]
I can probably get this to work by using osascript instead of pkexec on macOS like so:
```
#!/bin/sh
osascript -e "do shell script \"$*\" with administrator privileges"
```
Which should open the native macOS admin prompt...

The problem is that I don't have a Mac to test it on, so I need to setup a VM first. Might take a bit, in the meantime I might just make it throw an error.

**Comment by aklos** [comment]
Merged a fix that will use the osascript workaround for MacOS, and also gracefully fail if a child process is unable to spawn.

This fix will be in the 0.2.2 release.
